daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala

```diff
@@ -134,12 +214,32 @@ private[lf] final class CommandPreprocessor(
           choiceId,
           choiceArgument,
         )
-      case command.FetchCommand(templateId, coid) =>
-        // TODO https://github.com/digital-asset/daml/issues/10810
-        //  -- handle the case where templateId is an interface
+      case command.FetchCommand(identifier, coid) =>
+        val templateOrInterface = handleLookup(interface.lookupTemplateOrInterface(identifier))
+        val cid = valueTranslator.unsafeTranslateCid(coid)
+        templateOrInterface match {
+          case Left(template @ _) =>
+            speedy.Command.Fetch(identifier, cid)
+          case Right(interface @ _) =>
+            speedy.Command.FetchInterface(identifier, cid)
+        }
+      case command.FetchTemplateCommand(templateId, coid) => {
         discard[Ast.TemplateSignature](handleLookup(interface.lookupTemplate(templateId)))
```

Comment:
I am not sure the type parameter is very usefull in those `discard`

Reply:
Dropped it, thanks!